CRM/Utils/Token.php

```diff
@@ -990,6 +990,7 @@ public static function &replaceSubscribeInviteTokens($str) {
         'reset=1',
         TRUE, NULL, TRUE, TRUE
       );
+      $url = str_replace('&amp;', '&', $url);
```

Comment:
I think that instead of encoding and decoding the html entities, it would be simpler to just not encode them in the first place. Pass `FALSE` as the 5th argument to `CRM_Utils_System::url`.